packages/wdio-sauce-service/src/utils.js

```diff
@@ -0,0 +1,28 @@
+import { isW3C } from '@wdio/utils'
+
+export function makeCapabilityFactory(tunnelIdentifier, options) {
+    return capability => {
+        const isCapW3C = Boolean(
+            isW3C(capability) ||
+            capability['sauce:options'] ||
+            !(capability.platform || capability.version)
```

Comment:
No need to check for `platform` and `version`, it should be part of the `isW3C` check.
```suggestion
            capability['sauce:options'])
```

Reply:
The tests don't pass if that line isn't included.

I could add the checks to `isW3C`, but then `isW3C` would return true for an empty object.

Reply:
Which tests are failing? `isW3C` checks if any capabilities hint towards a W3C session. You are checking the absence of JSONWP capabilities which doesn't necessarily say that it is a W3C session.

Reply:
Tests which don't include `platformName` or `browserVersion`, i.e. https://github.com/webdriverio/webdriverio/blob/1f669ee23746b9e97b56a3a9baebe9ced94e5f75/packages/wdio-sauce-service/tests/launcher.test.js#L136

I agree that normally the absence of JSONWP capabilities doesn't necessarily say that it is a W3C session, but it seems like Sauce Labs uses only W3C or JSONWP, so in this case it is true?